### PR TITLE
Add Lewis and Wester benchmarks used in the GiNaC Paper

### PR DESCRIPTION
**Pull request by thilinarmtb**
Implemented the first five benchmarks in the Lewis and Wester benchmarks. Full list of the
benchmarks can be found here: http://www.bway.net/~lewis/cacomp.ps

### CODE REVIEW

**Comment by certik** [comment]
Cool! How are we doing compared to GiNaC? Can you post timings on your
machine?

File: benchmarks/lwbench_ginac.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+int main(int argc, char* argv[])
+{
+    std::cout << "Time for A : \t " << std::setw(15) << std::setprecision(9)  << std::fixed << A() << std::endl;
+    std::cout << "Time for B : \t " << std::setw(15) << std::setprecision(9)  << std::fixed << B() << std::endl;
+    std::cout << "Time for C : \t " << std::setw(15) << std::setprecision(9)  << std::fixed << C() << std::endl;
+    std::cout << "Time for D : \t " << std::setw(15) << std::setprecision(9)  << std::fixed << D() << std::endl;
+    std::cout << "Time for E : \t " << std::setw(15) << std::setprecision(9)  << std::fixed << E() << std::endl;
+    return 0;
+}
```
Comment (Sumith1896) [inline benchmarks/lwbench_ginac.cpp]: a new line

**Comment by thilinarmtb** [comment]
In my machine, CSymPy:
```
thilina@thilina-Inspiron:symengine$ ./benchmarks/lwbench 
Time for A : 	     0.007404512
Time for B : 	     0.001964378
Time for C : 	     0.015241531
Time for D : 	     0.000060285
Time for E : 	     0.000042570
thilina@thilina-Inspiron:symengine$ ./benchmarks/lwbench 
Time for A : 	     0.008269818
Time for B : 	     0.002458370
Time for C : 	     0.010606112
Time for D : 	     0.000052978
Time for E : 	     0.000041992
thilina@thilina-Inspiron:symengine$ ./benchmarks/lwbench 
Time for A : 	     0.007059362
Time for B : 	     0.001515069
Time for C : 	     0.010577192
Time for D : 	     0.000052903
Time for E : 	     0.000043133
thilina@thilina-Inspiron:symengine$ ./benchmarks/lwbench 
Time for A : 	     0.008222100
Time for B : 	     0.001428390
Time for C : 	     0.010610496
Time for D : 	     0.000052761
Time for E : 	     0.000042603
thilina@thilina-Inspiron:symengine$ ./benchmarks/lwbench 
Time for A : 	     0.007014080
Time for B : 	     0.001795352
Time for C : 	     0.010751368
Time for D : 	     0.000063036
Time for E : 	     0.000043127
```

GiNaC:
```
thilina@thilina-Inspiron:symengine$ ./lwbench_ginac 
Time for A : 	     0.029311898
Time for B : 	     0.001349352
Time for C : 	     0.014444329
Time for D : 	     0.000180880
Time for E : 	     0.000081537
thilina@thilina-Inspiron:symengine$ ./lwbench_ginac 
Time for A : 	     0.009148071
Time for B : 	     0.001312333
Time for C : 	     0.014654922
Time for D : 	     0.000103520
Time for E : 	     0.000074972
thilina@thilina-Inspiron:symengine$ ./lwbench_ginac 
Time for A : 	     0.011955404
Time for B : 	     0.001284139
Time for C : 	     0.014435863
Time for D : 	     0.000107480
Time for E : 	     0.000071206
thilina@thilina-Inspiron:symengine$ ./lwbench_ginac 
Time for A : 	     0.012732688
Time for B : 	     0.001269976
Time for C : 	     0.014633209
Time for D : 	     0.000100950
Time for E : 	     0.000071411
thilina@thilina-Inspiron:symengine$ ./lwbench_ginac 
Time for A : 	     0.011994654
Time for B : 	     0.001274375
Time for C : 	     0.014585881
Time for D : 	     0.000102307
Time for E : 	     0.000070820
```

File: benchmarks/lwbench_ginac.cpp (hunk 1 above)
Comment (thilinarmtb) [inline benchmarks/lwbench_ginac.cpp]: Done. Thanks !

**Comment by certik** [comment]
Ok, se we need to see why we are slower on the B benchmark.

**Comment by certik** [comment]
Anyway, this PR is +1 to merge as it is from me.

File: benchmarks/expand6_ginac.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// To complie on a debian system you need to install libginac-dev first
```
Comment (shivamvats) [inline benchmarks/expand6_ginac.cpp]: Should be 'compile'

Comment (thilinarmtb) [inline benchmarks/expand6_ginac.cpp]: Thanks !

**Comment by thilinarmtb** [comment]
I'll merge if the tests pass.

**Comment by Sumith1896** [comment]
Better if you can squash the last two commits.

**Comment by certik** [comment]
I think that's fine, you don't need to squash it.

**Comment by certik** [comment]
Thanks @thilinarmtb!

**Comment by thilinarmtb** [comment]
Thanks for merging !!

**Comment by Sumith1896** [comment]
Sorry if I was being troublesome.

**Pull request merged**